src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -549,6 +539,17 @@ function Set-AzConnectedKubernetes {
         #Endregion
 
         #Region Deal with configuration settings and protected settings
+        
+        # If the user does not provide proxy settings, or configuration settings, we shall use arc config of existing object
+        $userProvidedArcConfiguration = (
+            ($null -ne $InputObject) -and ($InputObject.ArcAgentryConfiguration.Length > 0) `
+            -and (-not ([string]::IsNullOrEmpty($HttpProxy))) `
+            -and (-not ([string]::IsNullOrEmpty($HttpsProxy))) `
```

Comment:
Why are we checking `https` config since this is never passed to ARM.

Reply:
Maybe because if we do provide it, it implies no other Arc config is required?

Reply:
It's true that the proxy configs will be saved in protected settings which is not going to be sent to ARM, but providing these in parameters means update to configuration settings and will send to configDP.